src/PythonProjectBootstrapper/EntryPoint.py

```diff
@@ -177,21 +191,345 @@ def _ExecuteOutputDir(
 
             execute_func = getattr(module, "Execute", None)
             if execute_func:
-                if execute_func(project_dir, output_dir, yes=yes) is False:
+                if execute_func(project_dir, tmp_dir, yes=yes) is False:
                     return
 
+    # generate project in temporary directory so we can avoid overwriting files without user approval
     cookiecutter(
         str(project_dir),
-        output_dir=str(output_dir),
+        output_dir=str(tmp_dir),
         config_file=str(configuration_filename) if configuration_filename is not None else None,
         replay=replay,
         overwrite_if_exists=True,
         accept_hooks=True,
     )
 
+    _CopyToOutputDir(tmp_dir=tmp_dir, output_dir=output_dir)
+    _DisplayPrompts(output_dir=output_dir)
+
 
 # ----------------------------------------------------------------------
+def _CreateManifest(generated_dir: Path) -> dict[Path, str]:
+
+    manifest_dict: dict[Path, str] = {}

```

Comment:
`PathEx.CreateRelativePath` returns a `PurePath`, so this type should be `dict[PurePath, str]`.